### ISSUE TITLE
Don't set the committee before checking it in `process_confirmed_block`.

### ISSUE THREAD

**Issue by afck** [linera-io/linera-protocol#2351]
[`process_confirmed_block` currently initializes the chain, including the committee](https://github.com/linera-io/linera-protocol/blame/9ee2d1d3e72e32ca2ab08d4c41845f0d42eeb575/linera-core/src/chain_worker/state/attempted_changes.rs#L261) before [checking the signatures based on that committee](https://github.com/linera-io/linera-protocol/blame/9ee2d1d3e72e32ca2ab08d4c41845f0d42eeb575/linera-core/src/chain_worker/state/attempted_changes.rs#L289).

**Comment by ma2bd** [comment]
Good catch. This is recent (removal of OpenChain dependency for confirmed blocks). I believe the fix is to put committees in an event stream (#365)

**Comment by ma2bd** [comment]
@afck Moving this to testnet2 due to the dependency to event streams. Perhaps we could add a `TODO(#2351)` in the code  now though?

**Comment by bart-linera** [comment]
#3787 is currently introducing similar issues in other places, I'll tag them with this issue's number and fix them all collectively in a follow-up PR.

**Comment by afck** [comment]
After #3787 the fix should now simply mean to strictly follow the rule again:

> The client should never put blobs into storage without first verifying the signatures of a certificate that uses that blob, and storing that certificate. This currently means executing the certificate (and all its ancestors), but after https://github.com/linera-io/linera-protocol/pull/3989 it only means adding that one block to the `unprocessed_blocks` set.

In addition, we can't consider a certificate from the chain itself to be proving its own chain description blob, unless we compare the committee to the admin chain.
